MonoGame.Framework/Platform/Graphics/Texture2D.OpenGL.cs

```diff
@@ -251,7 +251,7 @@ private void PlatformGetData<T>(int level, int arraySlice, Rectangle rect, T[] d
                 // Note: for compressed format Format.GetSize() returns the size of a 4x4 block
                 var pixelToT = Format.GetSize() / tSizeInByte;
                 var tFullWidth = Math.Max(this.width >> level, 1) / 4 * pixelToT;
-                var temp = new T[Math.Max(this.height >> level, 1) / 4 * tFullWidth];
+                var temp = GetDataPool<T>.GetArray(Math.Max(this.height >> level, 1) / 4 * tFullWidth);
```

Comment:
So this makes the code not thread safe...  but i suppose OpenGL never was?

@harry-cpp @mrhelmut ?

Reply:
GetData and SetData should only execute on UI thread ([line 228](https://github.com/MonoGame/MonoGame/blob/b1244f114a08b103d97c8fcec3e38f27dec93f24/MonoGame.Framework/Platform/Graphics/Texture2D.OpenGL.cs#L228)), so we can use this non thread safe pooling

I also thought about using ArrayPool.Shared, but it seemed that will be an overkill for this pr